File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        presum = 0
        maximum_sum = -float('inf')
        for i in range(len(nums)):
            if presum < 0:
                presum = nums[i]
            else:
                presum += nums[i]
            maximum_sum = max(maximum_sum, presum)
        
        return maximum_sum

File: 53-maximum-subarray/maximum-subarray.dart
class Solution {
  int maxSubArray(List<int> nums) {
    int current_sum = nums[0];
    int max_sum = nums[0];

    for(int i=1;i<nums.length; i++){
        current_sum = max(current_sum+ nums[i], nums[i]);
        max_sum = max(max_sum, current_sum);
    }
    return max_sum;
  }
}